### PR TITLE
master perf proper

### PR DESCRIPTION
**Pull request by janbiedermann**
#503

### CODE REVIEW

**Comment by schneems** [comment]
Sorry for the delay here. I think i'm ready to start looking at merging some of this stuff. Can you split this up into individual patches? 

Don't convert double quotes to single, performance difference is negligible and it messes up the git blame. My personal preference is double quotes everywhere so when they're mixed and right next to each other it looks fine.

for the `a_v_t_d` we already talked about this a ton here's some info. Using case is slower than the current method 

- https://github.com/rails/sprockets/pull/312#issuecomment-226369234.
- https://github.com/rails/sprockets/pull/417
- https://github.com/rails/sprockets/issues/383

I'm not sure what risks there are by changing the cache default numbers.

File: lib/sprockets/cache.rb [hunk 1]
```diff
@@ -67,7 +67,8 @@ def self.default_logger
     # cache - A compatible backend cache store instance.
     def initialize(cache = nil, logger = self.class.default_logger)
       @cache_wrapper = get_cache_wrapper(cache)
-      @fetch_cache   = Cache::MemoryStore.new(1024)
+      fetch_cache_size = cache.respond_to?(:max_size) ? cache.max_size / 1000 : 1024
```
Comment (schneems) [inline lib/sprockets/cache.rb]: Why divide by 1000? Why not just have them input the number that they want?

Also dividing an integer by an integer will give us a bad value  `1/1000 # => 0`

Comment (janbiedermann) [reply lib/sprockets/cache.rb]: i have new patches, need to cleanup and a new pull request will follow, maybe next week.

**Comment by janbiedermann** [comment]
i close this, i created a opal-webpack-loader instead

**Pull request closed without merging**